tests/rptest/services/transform_verifier_service.py

```diff
@@ -0,0 +1,332 @@
+        self._await_ready(node)
+
+        # Because the above command was run with `nohup` we can't be sure that
+        # it is the one who actually replied to the `await_ready` calls.
+        # Check that the PID we just launched is still running as a confirmation
+        # that it is the one.
+        self._assert_running(node)
+
+    def wait_node(self, node, timeout_sec=None):
+        """
+        Wait until the status endpoint reports that it is done
+        """
+        wait_until(
+            lambda: self.config.is_done(self._get_status_node(node)),
+            timeout_sec=timeout_sec or 5,
+            backoff_sec=0.5,
+            err_msg=
+            f"Timed out for transform verifier to complete {self.who_am_i()}")
+        return True
+
+    def stop_node(self, node):
+        """
+        Stop the process on this node
+        """
+        self.redpanda.logger.info(f"{self.__class__.__name__}.stop")
+        if self._pid is None:
+            return
+
+        # Attempt a graceful stop
+        self._execute_cmd(node, "stop")
+        try:
+            wait_until(lambda: not node.account.exists(f"/proc/{self._pid}"),
+                       timeout_sec=10,
+                       backoff_sec=0.5)
+            self._pid = None
+            return
+        except TimeoutError as e:
+            self.logger.warn("gracefully stopping service failed: {e}")
+
+        # Gracefully stop did not work, try a hard kill
+        self.logger.debug(f"Killing pid {self._pid}")
+        try:
+            node.account.signal(self._pid, signal.SIGKILL, allow_fail=False)
+        except RemoteCommandError as e:
+            if b"No such process" not in e.msg:
+                raise
+        self._pid = None
+
+    def get_status(self):
+        status = None
+        for node in self.nodes:
+            node_status = self._get_status_node(node)
+            if not status:
+                status = node_status
+            else:
+                status = status.merge(node_status)
+        return status
+
+    def _await_ready(self, node):
+        """
+        Wait for the remote processes http endpoint to come up
+        """
+        wait_until(
+            lambda: self._is_ready(node),
+            timeout_sec=20,
+            backoff_sec=.25,
+            err_msg=
+            f"Timed out waiting for status endpoint {self.who_am_i()} to be available"
+        )
+
+    def _execute_cmd(self, node, cmd):
+        """
+        Execute a command on all nodes.
+
+        Supported Commands:
+        - "stop"
+        """
+        self.logger.debug(f"Invoking remote path {cmd} on {node.name}")
+        r = requests.get(self._remote_url(node, cmd), timeout=10)
+        r.raise_for_status()
+
+    def _remote_url(self, node, path):
+        return f"http://{node.account.hostname}:{self.remote_port}/{path}"
+
+    def _is_ready(self, node):
+        try:
+            r = requests.get(self._remote_url(node, "status"), timeout=10)
```

Comment:
nit: there are several methods that bottom out at a `GET /status` request, might be nice to consolidate

Reply:
Done